sdk/lib/rtl/dbgbuffer.c

```diff
@@ -14,30 +15,80 @@
 
 /* FUNCTIONS *****************************************************************/
 
+PVOID
+NTAPI
+RtlpDebugBufferCommit(_Inout_ PRTL_DEBUG_INFORMATION Buffer,
+                     _In_ SIZE_T Size)
+{
+    ULONG Remaining = Buffer->CommitSize - Buffer->OffsetFree;
+    PVOID Result;
+    NTSTATUS Status;
+
+    if (Size > 0x7ffffff)
+        return NULL;
+
+    if (Remaining < Size)
+    {
+        PVOID Buf;
+        SIZE_T CommitSize;
+
+        Buf = (PVOID)((ULONG_PTR)Buffer->ViewBaseClient + Buffer->CommitSize);
+        CommitSize = Size - Remaining;
+
+        /* this is not going to end well.. */
+        if (CommitSize > 0x7ffffff)
+            return NULL;
+
+        Status = NtAllocateVirtualMemory(NtCurrentProcess(), (PVOID*)&Buf, 0, &CommitSize, MEM_COMMIT, PAGE_READWRITE);
+        if (!NT_SUCCESS(Status))
+            return NULL;
+
+        Buffer->CommitSize += CommitSize;
+        Remaining = Buffer->CommitSize - Buffer->OffsetFree;
+        /* Sanity check */
+        ASSERT(Remaining >= Size);
+        if (Remaining < Size)
+            return NULL;
+    }
+
+    Result = (PBYTE)Buffer->ViewBaseClient + Buffer->OffsetFree;
+    Buffer->OffsetFree += Size;
+
+    return Result;
+}
+
+
 /*
  * @unimplemented
  */
 PRTL_DEBUG_INFORMATION
 NTAPI
-RtlCreateQueryDebugBuffer(IN ULONG Size,
-                          IN BOOLEAN EventPair)
+RtlCreateQueryDebugBuffer(_In_ ULONG Size,
+                          _In_ BOOLEAN EventPair)
 {
     NTSTATUS Status;
     PRTL_DEBUG_INFORMATION Buf = NULL;
-    SIZE_T ViewSize = 100 * PAGE_SIZE;
+    SIZE_T AllocationSize = Size ? Size : 0x400000;
```

Comment:
0x400000 == 0x400 * PAGE_SIZE

Reply:
I simply chose whatever came out of the test result, but this'll work as well :)